### PR TITLE
[#12502][fix] Export softmax stats with the real scale so SM90 MLA chunked prefill works

### PR DESCRIPTION
**Pull request by akhoroshev**
## Description

MLA chunked prefill was disabled on SM90 by #13904 because merged chunks were
numerically wrong (max diff ~0.45). This fixes the underlying kernel bug and re-enables
the path.

The Hopper warp-specialized context FMHA keeps the raw BMM1 max in `p_max` and applies
the softmax scale only inside `exp2f`. When exporting the softmax stats it rescaled that
max by `1/sqrt(head_dim)`:

```cpp
saver.store(p_sum, p_max, sqrtf(params.d), q_step_idx * STEP_Q, valid_run);
...
values = p_max[lane % 2] / sqrt_d;
```

but the scale the softmax actually used is `1/(sqrt(head_dim) * q_scaling)`
(`fmhaRunner.cpp`: `inv_sqrt_scale = 1.f / (sqrtf(headSize) * qScaling)`), so the
`q_scaling` factor was dropped. The companion sum is dimensionless — it is
`sum(exp(logit - max))` — and was therefore already correct, so the two exported columns
disagreed by exactly `q_scaling`.

`forward_context_with_chunked_prefill` merges chunks with `exp(max_i - max)` together
with that sum, so for any model with `q_scaling != 1` (YaRN mscale, i.e. DeepSeek-V3
style MLA) chunks were weighted incorrectly.

The fix passes the scale the softmax actually used (`softmax.scale_bmm1_`, undoing the
`log2(e)` factor of the base-2 exponent trick) and multiplies instead of dividing, then
drops the `get_sm_version() >= 100` gate.

Scope: only sm90 kernels compile `warpspec/compute.h` (512 generated `.cu` files;
sm89/sm100/sm120 compile none of it), and the other MLA shapes take the ampere-style
path where the scale is applied to the logits directly (`hmul2(acc, scale_bmm1_)`) and
the exported max is already in the right units. The stats are consumed only by
`invokeMergeAttnWithSoftmax` in the MLA chunked-prefill merge. The non-chunked prefill
path does not request stats and is bit-identical before and after the change.

Measurements on H100, DeepSeek-V3-style MLA with `q_scaling = 0.7713` (YaRN mscale):

- exported max vs a hand-computed reference: off by exactly `1/q_scaling = 1.29647`
  before, matches to 5 decimals after;
- 50k-token prompt, 16k chunks, attention layers only, against a full non-chunked
  prefill of the same prompt: max relative L2 error per position **0.51 -> 0.23**,
  min cosine similarity **0.862 -> 0.973** — identical to the cached-KV fallback, and
  stable across 8k and 16k chunk sizes;
- end to end on a 40-layer model: top-1 agreement **0.9711**, `|delta mean NLL|`
  **0.0025** versus non-chunked prefill.

## Test Coverage

- `tests/unittest/_torch/attention/test_attention_mla.py::test_mla_context_softmax_stats_units`
  (new) — runs the real context MLA attention with a `softmax_stats_tensor` and
  `latent_cache=None` (which skips RoPE and the KV-cache write, so the kernel attends
  over exactly the q/k passed in), then checks **both** exported stats columns against a
  causal PyTorch reference. Parametrized with `q_scaling=1.0` (control, unaffected by the
  bug) and `q_scaling=0.7713206` (regression).
  Verified to actually catch the bug: rebuilt with the kernel change reverted, the control
  passes while the regression case fails with relative difference `0.2287 = 1 - q_scaling`.
- `tests/unittest/_torch/attention/test_attention_mla.py::test_mla_chunked_prefill_dispatch_by_sm`
  (updated) — now expects the chunked path on every SM.
- `tests/unittest/_torch/attention/test_attention_mla.py` as a whole is picked up on
  Hopper by the `unittest/_torch/attention` entry in
  `tests/integration/test_lists/test-db/l0_h100.yml`, so both tests run in CI on the
  architecture this fix targets.

Note on the pre-existing coverage gap this bug slipped through:
`cpp/tests/unit_tests/kernels/mlaChunkedPrefillTest.cu` exercises
`invokeMergeAttnWithSoftmax` on softmax stats it generates itself and never runs the FMHA
kernel, so it validates the consumer but not the producer, and it does not model
`q_scaling` at all.

## Dev Engineer Review

- MLA softmax maxima now use `softmax.scale_bmm1_`, including `q_scaling`.
- `Softmax_saver_tma::store` now accepts the active softmax scale.
- The `EXP2F_OPTIMIZATION` path converts the scale with `ln(2)`.
- Chunked MLA prefill no longer applies the SM-version gate.
- Verify scale units and chunk merging across optimization modes.

## QA Engineer Review

- Modified `tests/unittest/_torch/attention/test_attention_mla.py`.
- Added parameterized softmax-statistics coverage for `q_scaling=1.0` and `q_scaling=0.7713206`.
- Added chunked-prefill output coverage against the single-shot cached-KV path.
- Updated dispatch coverage for SM90, SM99, and SM100.
- The test file is listed in `l0_rtx_pro_6000.yml`, `l0_gb10.yml`, and `l0_gb202.yml`.
- Test execution results were not supplied.
- Coverage verdict: **sufficient**.

## Per-File QA Perspective

- `cpp/kernels/fmha_v2/src/fmha/hopper/fragment.h`: Verify that callers pass the active softmax scale and that exported maxima use the expected units.
- `cpp/kernels/fmha_v2/src/fmha/warpspec/compute.h`: Verify standard and `EXP2F_OPTIMIZATION` statistics export with non-default `q_scaling`.
- `tensorrt_llm/_torch/attention/mla.py`: Verify chunked MLA prefill dispatches to `forward_context_with_chunked_prefill` and that cached-KV fallback behavior remains unchanged.
- `tests/unittest/_torch/attention/test_attention_mla.py`: Covers causal softmax-statistics units, chunked versus single-shot output, and SM-based dispatch. The file is listed in all three applicable CI test lists.

### CODE REVIEW

**Comment by bobboli** [comment]
Adding @jmydurant for review.

File: cpp/kernels/fmha_v2/src/fmha/warpspec/compute.h [hunk 1]
```diff
@@ -535,7 +535,16 @@ struct Compute
                 {
                     using Mma_tile = typename Traits_p::template Mma_tile<Cta_tile_o>;
                     fmha::Softmax_saver_tma<Cta_tile_o, Mma_tile> saver(params, head_info);
-                    saver.store(p_sum, p_max, sqrtf(params.d), q_step_idx * STEP_Q, valid_run);
+                    // p_max holds the raw BMM1 max, so export it with the same
+                    // scale the softmax used. scale_bmm1_ carries an extra
+                    // log2(e) when the base-2 exponent trick is on.
+                    float softmax_scale = reinterpret_cast<float const&>(softmax.scale_bmm1_);
```
Comment (jmydurant) [inline cpp/kernels/fmha_v2/src/fmha/warpspec/compute.h]: The scale conversion looks correct for the EXP2-enabled MLA path.

  For `EXP2F_OPTIMIZATION == false`, the logits have already been scaled in `unpack()`, and any ALiBi/softcapping has been applied before computing `p_max`. The maximum is therefore already
  in the units required by the consumer and should be exported directly. Multiplying it by `scale_bmm1_` again would make it inconsistent with `p_sum`.

  Could we initialize the export scale to `1.f` and only convert it in the EXP2 branch?

  ```cpp
  float softmax_scale = 1.f;
  if constexpr (Kernel_traits::EXP2F_OPTIMIZATION)
  {
      constexpr float kLn2 = 0.693147180559945309417f;
      softmax_scale =
          reinterpret_cast<float const&>(softmax.scale_bmm1_) * kLn2;
  }
  ```

  `p_sum` needs no additional BMM1 scale or base conversion: the EXP2 path already computes the same exponential sum as the natural-exponential formulation.

  The previous `/ sqrt_d` also appears incorrect for the non-EXP2 path, so this is a pre-existing gap in the shared exporter. Since non-MLA kernels also instantiate this path, could we add a
  statistics regression test with ALiBi or softcapping enabled?
